src/ResourceManagement/ResourceManager/GenericResource/GenericResourceImpl.cs

```diff
@@ -97,6 +98,17 @@ public void prepareDefaultValue()
         public async override Task<IGenericResource> CreateResourceAsync(CancellationToken cancellationToken)
         {
             prepareDefaultValue();
+            if (apiVersion == null)
+            {
+                string resourceId = ResourceUtils.ConstructResourceId(Manager.Inner.SubscriptionId, ResourceGroupName, resourceProviderNamespace, resourceType, Name, parentResourceId);
+                apiVersion = ResourceUtils.DefaultApiVersionFromResourceId(
+                    resourceId, 
+                    await Manager.Providers.GetByNameAsync(resourceProviderNamespace, cancellationToken));
+                if (apiVersion == null)
+                {
```

Comment:
This is not an argument exception. At least, argument is not passed as parameter here.

Reply:
The argument exception will throw when it cannot find API version when using resource provider/resource type/resource id from method signature. I think it would be ArgumentException. Otherwise lets use InvalidOperationException?